repose-aggregator/components/filters/keystone-v2-filter/src/main/scala/org/openrepose/filters/keystonev2/KeystoneV2Authorization.scala

```diff
@@ -59,19 +56,31 @@ object KeystoneV2Authorization extends LazyLogging {
     }
   }
 
-  val handleFailures: PartialFunction[Try[Unit.type], KeystoneV2Result] = {
-    case Failure(e: InvalidTenantException) => Reject(SC_UNAUTHORIZED, Some(e.getMessage))
-    case Failure(e: UnauthorizedEndpointException) => Reject(SC_FORBIDDEN, Some(e.getMessage))
-    case Failure(e: UnparseableTenantException) => Reject(SC_UNAUTHORIZED, Some(e.getMessage))
+  // TODO: Switch to getRequestTenants (plural) and include all tenants extracted. This is a behavior change.
+  def getRequestTenant(config: ValidateTenantType, request: HttpServletRequestWrapper): String = {
+    Option(config).flatMap(validateTenantConfig =>
+      (validateTenantConfig.getUriExtractionRegexAndHeaderExtractionName.asScala.toStream map {
+        case headerExtraction: HeaderExtractionType =>
+          request.getPreferredSplittableHeaders(headerExtraction.getValue).asScala.headOption
```

Comment:
this seems super broke. you can only possibly match on the first header of a given quality

Reply:
That is exactly what it does. You'll also notice that I turn the list of extractions into a `Stream`, and then terminate the stream with a `find`. The big idea is that the first tenant that matches will be returned.

This is the single-tenant approach consistent with current behavior, but it will change with the next story and PR.

Reply:
Yeah, i was thinking of the other header of tenants, not the ones to match. derp.